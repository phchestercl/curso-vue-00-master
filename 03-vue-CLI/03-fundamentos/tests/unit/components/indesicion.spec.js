/**
 * Hacer match con snapshot
 * 
 */

import { shallowMount, mount } from '@vue/test-utils'
import Indesicion from '@/components/Indecision'
 describe('Test Indesicion',()=>{
     let wrapper;
     let consoleLogSpy;
     /** Mock de la funcion fetch */
     global.fetch = jest.fn( ()=>Promise.resolve({
         json:()=>Promise.resolve({
                'answer': 'yes',
                'forced': false,
                'image': 'https://yesno.wtf/assets/yes/2.gif'
         })
     }) )


     beforeEach(()=>{
         wrapper=shallowMount(Indesicion)
         consoleLogSpy=jest.spyOn( console ,'log')
         /** limpiar mocks */
         jest.clearAllMocks()
     })
     test('Prueba de snapshot del componente',()=>{
         expect(wrapper.html()).toMatchSnapshot()
     })
     test('Describir en el Input no debe disparar nada (console.log)',async()=>{
        /* console.log(wrapper) */
        const getAnswerSpy= jest.spyOn(wrapper.vm,'getAnswer')
        const input=wrapper.find('input')
        await input.setValue('Hola Mundo')
        /* expect(consoleLogSpy).toHaveBeenCalled() */
        expect(consoleLogSpy).toHaveBeenCalledTimes(1)
        //revisamos que la funcion no haya sido llamado
        expect(getAnswerSpy).toHaveBeenCalledTimes(0)
       
     })
     test('Al escribir el ? disparar getAnswer',async()=>{
         /** Aqui solo nos ineteresa saber si el getAnwser fue llamado no estamos probando como está funcionando el getAnswer*/
         const getAnswerSpy=jest.spyOn(wrapper.vm,'getAnswer');
         const input=wrapper.find('input')
         await input.setValue('Está funconado ok?')
         
         expect(getAnswerSpy).toHaveBeenCalled()

     })
     test('Pruebas en getAnswer', async()=>{
        await wrapper.vm.getAnswer()
        /* console.log(wrapper.vm.img)
        console.log(wrapper.vm.answer) */
        const img =wrapper.find('img')
        expect(img).toBeTruthy()
        expect(wrapper.vm.answer).toBe('Si!!')
     })
     test('Pruebas en getAnswer -Fallo en Api',async()=>{

        fetch.mockImplementationOnce(()=>Promise.reject('API is Down'));
        await wrapper.vm.getAnswer()
        const img =wrapper.find('img')
        expect(img.exists()).toBeFalsy()
        expect(wrapper.vm.answer).toBe('No se pudo obtener data del API')
     })
 })