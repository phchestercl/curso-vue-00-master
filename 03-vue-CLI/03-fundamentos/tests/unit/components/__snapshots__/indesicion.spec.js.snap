// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Indesicion Prueba de snapshot del componente 1`] = `
<div>
  <!--v-if-->
  <div class="bg-dark"></div>
  <div class="indecision-container"><input name="" id="" placeholder="Hazme una pregunta">
    <p> Recuerda terminar con un signo de interrogaci√≥n (?)</p>
    <!--v-if-->
  </div>
</div>
`;
